app/models/releases/step/run.rb

```diff
@@ -118,6 +117,7 @@ class Releases::Step::Run < ApplicationRecord
 
     event(:finish) do
       after { event_stamp!(reason: :finished, kind: :success, data: stamp_data) }
+      after { finalize_release }
```

Comment:
this should be `after_commit`, not just `after`, no? Although in this case there is not a lot that can fail the transaction, we should be consistent. I do not remember why we had that event stamp in the after block, instead of after commit.